pkg/synchromanager/clustersynchro/resource_synchro.go

```diff
@@ -400,3 +395,22 @@ func (synchro *ResourceSynchro) deleteResource(obj runtime.Object) error {
 func (synchro *ResourceSynchro) Status() clusterv1alpha2.ClusterResourceSyncCondition {
 	return synchro.status.Load().(clusterv1alpha2.ClusterResourceSyncCondition)
 }
+
+func ErrorHandlerForResourceSynchro(synchro *ResourceSynchro) informer.WatchErrorHandler {
+	return func(_ *informer.Reflector, err error) {
+		status := clusterv1alpha2.ClusterResourceSyncCondition{
+			Status:             clusterv1alpha2.SyncStatusSyncing,
+			Reason:             "",
+			LastTransitionTime: metav1.Now().Rfc3339Copy(),
+		}
+		if err != nil {
+			status = clusterv1alpha2.ClusterResourceSyncCondition{
+				Status:             clusterv1alpha2.SyncStatusError,
+				Reason:             "ResourceWatchFailed",
+				Message:            err.Error(),
+				LastTransitionTime: metav1.Now().Rfc3339Copy(),
+			}
+		}
+		synchro.status.Store(status)
```

Comment:
Do we need to call the `DefaultWatchErrorHandler ` again so that it may be more beneficial when analyzing the logs separately?

Reply:
Yes, good idea.